tests/unit/sort.tcl

```diff
@@ -204,6 +204,11 @@ foreach command {SORT SORT_RO} {
         assert_equal [lsort -real $floats] [r sort mylist]
     }
 
+    test "SORT returns an empty array if the list is empty" {
+        r flushdb
+        r sort mylist
```

Comment:
Running this with gdb, vectorlen for both `sort mylist` and `sort mylist by *` both showed vectorlen with a value of zero. I didn't spend too much time validating, but I couldn't quite trace the path that caused this to be -1.

Reply:
My bad, it's not vectorlen that underflows, it's start/end.

Reply:
I see! That makes more sense.

Instead of adding a new special return flow, I think it might be a bit cleaner to avoid calling pqsort when start or end is negative (which represents the case when length is 0 or 1, which doesn't require sorting anyways).